File: src/main/java/kdh/boardproject/dto/board/BoardDto.java
package kdh.boardproject.dto.board;

import kdh.boardproject.dto.category.CategoryListDto;
import kdh.boardproject.dto.user.UserRequestDto;
import kdh.boardproject.entity.Board;
import kdh.boardproject.entity.BoardComment;
import kdh.boardproject.entity.Category;
import kdh.boardproject.entity.User;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class BoardDto {
    private Long idx;
    private String title;
    private String content;
    private LocalDateTime createdAt;
    private UserRequestDto user;
    private CategoryListDto category;

    public BoardDto(Board board){
        this.idx = board.getIdx();
        this.title = board.getTitle();
        this.content = board.getContent();
        this.createdAt = board.getCreatedAt();
        this.user = new UserRequestDto(board.getUser());
        this.category = new CategoryListDto(board.getCategory());
    }
}

File: src/main/java/kdh/boardproject/dto/board/BoardDto.kt
package kdh.boardproject.dto.board

import kdh.boardproject.dto.category.CategoryListDto
import kdh.boardproject.dto.user.UserRequestDto
import kdh.boardproject.entity.Board
import java.time.LocalDateTime


class BoardDto(board: Board) {
     val idx: Long?
     val title: String
     val content: String
     val createdAt: LocalDateTime?
     val user: UserRequestDto
     val category: CategoryListDto

    init {
        idx = board.idx
        title = board.title
        content = board.content
        createdAt = board.createdAt
        user = UserRequestDto(board.user)
        this.category = CategoryListDto(board.category)
    }
}